app/models/mixins/owner_mixin.rb

```diff
@@ -0,0 +1,5 @@
+module OwnerMixin
+  def set_owner
+    self.owner = ManageIQ::API::Common::Request.current.user.username
+  end
+end
```

Comment:
Using `ActiveSupport::Concern` you can DRY up these changes by moving `validates` and `before_validation` logic into the mixin using an `included` block.

You can see an example here: https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/dialog_mixin.rb#L2-L6

Reply:
Cool! Didn't know this was the right use-case for it. Updated!